mod.ts

```diff
@@ -99,10 +160,43 @@ export function denoPlugin(options: DenoPluginOptions = {}): esbuild.Plugin {
             return portableLoad(url, options);
         }
       }
+      async function nodePackage(
+        args: esbuild.OnLoadArgs,
+      ): Promise<esbuild.OnLoadResult | null | undefined> {
+        let ref: NpmPackageReference | undefined = npmModulesCache
+          .get(args.path);
+
+        if (!ref) {
+          ref = npmPackageReference(args.path);
+          npmModulesCache.set(args.path, ref);
+        }
+        const specifier = ref.name;
+        const version = ref.versionReq;
+
+        
+        const moduleDirPath =
+          `${npmCache}/registry.npmjs.org/${specifier}/${version}`;
+        //Make Sure the Package is cached at NPM Cache Else It will be downloaded See Func Definition..
+        await checkExistNpmMod(args.path, moduleDirPath);
+        const packageJson = JSON.parse(
+          await Deno.readTextFile(
+            `${moduleDirPath}/package.json`,
+          ),
+        );
+
+        let file = packageJson.main ?? "index.js";
+        if (ref.subPath) {
+          file = `${ref.subPath}.js`;
+        }
+        return {
+          contents: await Deno.readTextFile(`${moduleDirPath}/${file}`),
+        };
+      }
       build.onLoad({ filter: /.*\.json/, namespace: "file" }, onLoad);
       build.onLoad({ filter: /.*/, namespace: "http" }, onLoad);
       build.onLoad({ filter: /.*/, namespace: "https" }, onLoad);
       build.onLoad({ filter: /.*/, namespace: "data" }, onLoad);
+      build.onLoad({ filter: /.*/, namespace: "node" }, nodePackage);
```

Comment:
This should probably be namespace `npm` - there could be an import like `import fs from "node:fs"` which are actual built-in Node modules

Reply:
Yeah that was a silly mistake , i already said that im will change in the next commit and yeah i forget😅.. and i wil come with proper new commit..